### PR TITLE
[ACA-3692] - Add completed date/due date/started date filter

### PR DESCRIPTION
**Pull request by SilviuCPopa**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ACA-3692

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by VitoAlbano** [comment]
Please fix the unit tests

File: lib/process-services-cloud/src/lib/process/process-filters/components/edit-process-filter-cloud.component.ts [hunk 1]
```diff
@@ -164,7 +165,14 @@ export class EditProcessFilterCloudComponent implements OnInit, OnChanges, OnDes
 
     getFormControlsConfig(processFilterProperties: ProcessFilterProperties[]): any {
         const properties = processFilterProperties.map((property: ProcessFilterProperties) => {
-            return { [property.key]: property.value };
+            if (!property.rangeKeys) {
```
Comment (popovicsandras) [inline lib/process-services-cloud/src/lib/process/process-filters/components/edit-process-filter-cloud.component.ts:168]: After the changes what we discussed, do we still need this dirty change? :P

File: lib/process-services-cloud/src/lib/process/process-filters/models/process-filter-cloud.model.ts [hunk 2]
```diff
@@ -57,7 +61,11 @@ export class ProcessFilterCloudModel {
             this.lastModified = obj.lastModified || null;
             this.lastModifiedTo = obj.lastModifiedTo || null;
             this.lastModifiedFrom = obj.lastModifiedFrom || null;
+            this.completedFrom = obj.completedFrom || null;
```
Comment (popovicsandras) [inline lib/process-services-cloud/src/lib/process/process-filters/models/process-filter-cloud.model.ts:64]: Is it the last version, what we discussed? 🤔

**Review by popovicsandras — CHANGES_REQUESTED** [review]

File: lib/process-services-cloud/src/lib/process/process-filters/models/process-filter-cloud.model.ts [hunk 3]
```diff
@@ -62,10 +72,71 @@ export class ProcessFilterCloudModel {
...
+            this.startTo = obj._startTo || null;
+            this.completedDateType = obj.completedDateType || null;
+            this.startedDateType = obj.startedDateType || null;
+            this.completedFrom = obj._completedFrom || null;
+            this.completedTo = obj._completedTo || null;
+            this.completedDate = obj.completedDate || null;
         }
     }
+
+    set completedFrom(completedFrom: string) {
```
Comment (popovicsandras) [inline lib/process-services-cloud/src/lib/process/process-filters/models/process-filter-cloud.model.ts:85]: Duplication type #1

File: lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filter-cloud.component.ts [hunk 4]
```diff
@@ -170,18 +170,22 @@ export class EditTaskFilterCloudComponent implements OnInit, OnChanges, OnDestro
 
     getFormControlsConfig(taskFilterProperties: TaskFilterProperties[]): any {
         const properties = taskFilterProperties.map((property: TaskFilterProperties) => {
-            if (!property.rangeKeys) {
-                return { [property.key]: property.value };
+            if (!!property.attributes) {
```
Comment (popovicsandras) [inline lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filter-cloud.component.ts:173]: Duplication type #2

**Review by popovicsandras — APPROVED** [review]
Given the close deadline, I approve it, but please for the marked duplication types create a separate Jira ticket in the next Technical Debt/Maintenance epic and assign to yourself and put into the sprint. (I think it is quite clear what I mean by the marked duplications, otherwise, let's discuss)

**Pull request merged**